app/src/ui/lib/list/list.tsx

```diff
@@ -89,11 +89,11 @@ interface IListProps {
   readonly rowHeight: number | ((info: { index: number }) => number)
 
   /**
-   * The currently selected row index. Used to attach a special
-   * selection class on that row container as well as being used
+   * The currently selected rows indexes. Used to attach a special
+   * selection class on those row's containers as well as being used
    * for keyboard selection.
    */
-  readonly selectedRow: number
+  readonly selectedRows: number[]
```

Comment:
I think it'd be nice if we could apply the same approach we took to the callbacks here and have this be a type union. Something similar to

```ts
type Selection = SingleSelection | RangeSelection | MultiSelection

type SingleSelection = {
  kind: 'single'
  row: number
}

type RangeSelection = {
  kind: 'range'
  start: number
  end: number
}

type MultiSelection = {
  kind: 'multi'
  rows: number[]
}

// [...]
readonly selection: Selection
```

What do you think @icosamuel?

Reply:
This change would imply rewriting every place where the selection is used for processing. The biggest part of the PR was to change those processes from working with single elements to an array of elements. I think we should add something like a dynamic property to build the array of selected items from the suggested union type property before handling them everywhere in the class; or leave this responsibility to the using components like `filter-list`or `file-list`

Reply:
@niik If we go with that, what about renaming `start` and `end` to `first` and `last` to clarify the wording?

Reply:
> This change would imply rewriting every place where the selection is used for processing.

Yeah, you're right. I think it could end being a nice abstraction but perhaps we shouldn't overload this initial PR with that just yet. An array can, after all, represent all the various selection modes while only giving a slight penalty to the range mode which we won't be using until the comparison work lands.

Let's punt on it in this PR.

> @niik If we go with that, what about renaming start and end to first and last to clarify the wording?

I don't have strong feelings on that. I think `start` and `end` is a fairly widespread convention.